addons/sale/models/sale.py

```diff
@@ -1046,20 +1046,25 @@ def _prepare_add_missing_fields(self, values):
                     res[field] = line._fields[field].convert_to_write(line[field], line)
         return res
 
-    @api.model
-    def create(self, values):
-        if values.get('display_type', self.default_get(['display_type'])['display_type']):
-            values.update(product_id=False, price_unit=0, product_uom_qty=0, product_uom=False, customer_lead=0)
-
-        values.update(self._prepare_add_missing_fields(values))
-        line = super(SaleOrderLine, self).create(values)
-        if line.product_id and line.order_id.state == 'sale':
-            msg = _("Extra line with %s ") % (line.product_id.display_name,)
-            line.order_id.message_post(body=msg)
-            # create an analytic account if at least an expense product
-            if line.product_id.expense_policy not in [False, 'no'] and not self.order_id.analytic_account_id:
-                self.order_id._create_analytic_account()
-        return line
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
```

Comment:
My 2 cents here: if you used `for values in vals_list:` and and avoided reformatting, we could more easily review the diff with 'Hide whitespace changes'